### PR TITLE
import neptune instead of import neptune.new

### LINKED ISSUE #16887: neptune.new deprecation warning with NeptuneLogger
### Bug description

When using the latest versions of pytorch-lightning and neptune-client from conda-forge, I get a deprecation warning.

### How to reproduce the bug

```python
Simply creating a NeptuneLogger should give the error already.
```

### Error messages and logs

```bash
/home/micromamba/lib/python3.10/site-packages/pytorch_lightning/loggers/neptune.py:39: NeptuneDeprecationWarning: You're importing the Neptune client library via the deprecated `neptune.new` module, which will be removed in a future release. Import directly from `neptune` instead.
  from neptune import new as neptune
```

### Environment

<details>
  <summary>Current environment</summary>

```
- Lightning Component: NeptuneLogger
- neptune-client: 1.0.1
- pytorch-lightning: 1.9.3
- PyTorch: 1.13.1
- Python: 3.10
```

</details>

### More info

https://docs.neptune.ai/setup/neptune-client_1-0_release_changes/

cc @justusschock @awaelchli @Blaizzy

### PR DESCRIPTION
**Pull request by wouterzwerink**
## What does this PR do?

Updates the importing of neptune according to https://docs.neptune.ai/setup/neptune-client_1-0_release_changes/
This probably breaks NeptuneLogger with neptune-client<1.0.0. Should neptune-client>=1.0.0 be added as a requirement?

Fixes #16887

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: docs/source-pytorch/visualize/supported_exp_managers.rst [hunk 1]
```diff
@@ -68,13 +68,13 @@ To use `Neptune.ai <https://neptune.ai/>`_ first install the neptune package:
...
 
-    pip install neptune-client
+    pip install neptune
 
 or with conda:
 
 .. code-block:: bash
 
-    conda install -c conda-forge neptune-client
+    conda install -c conda-forge neptune
```
Comment (wouterzwerink) [inline docs/source-pytorch/visualize/supported_exp_managers.rst:77]: The conda-forge package has not been renamed yet (I believe this is in progress), so this currently does not work.

File: src/lightning/pytorch/loggers/neptune.py [hunk 2]
```diff
@@ -55,13 +55,13 @@ class NeptuneLogger(Logger):
...
 
-        pip install neptune-client
+        pip install neptune
 
     or conda:
 
     .. code-block:: bash
 
-        conda install -c conda-forge neptune-client
+        conda install -c conda-forge neptune
```
Comment (wouterzwerink) [inline src/lightning/pytorch/loggers/neptune.py:64]: The conda-forge package has not been renamed yet (I believe this is in progress), so this currently does not work.

File: docs/source-pytorch/visualize/supported_exp_managers.rst (hunk 1 above)
Comment (carmocca) [reply docs/source-pytorch/visualize/supported_exp_managers.rst:77]: Sorry. Can you revert those changes then?

**Comment by carmocca** [comment]
cc @Blaizzy for review too

**Comment by Borda** [comment]
> cc @Blaizzy for review too

lgtm :)

**Comment by Blaizzy** [comment]
Hi @wouterzwerink and @carmocca 

Thanks for the initiative!

However, this PR solves part of the problem but not all.

There are a few more issues precisely:
- Users still using the legacy client(<=1.0.0) will have their workflows broken.
- The new client no longer supports automatically converting unsupported data types to strings. (Docs: https://docs.neptune.ai/api/utils/#stringify_unsupported)

We are preparing a PR to address the breaking changes and ensure backward compatibility.

So, could you please revert this PR?

Here is the draft PR we have been working on for a few weeks:
https://github.com/Lightning-AI/lightning/pull/16761/files

**Comment by wouterzwerink** [comment]
@Blaizzy sure, reverting in #16898.

**Comment by carmocca** [comment]
@Blaizzy you can incorporate those changes and backwards compatibility in #16761 after rebasing this PR's changes. Without needing to revert

**Comment by Blaizzy** [comment]
@carmocca sure, no problem!

**Comment by Blaizzy** [comment]
Thanks, guys!

I really appreciate the initiative, @wouterzwerink!

**Pull request merged**